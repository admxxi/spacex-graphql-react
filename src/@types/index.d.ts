/* eslint-disable */
export interface IPost {
  id: number
  title: string
  body: string
}

export type Country = {
  code: !number
  name: !string
  emoji: !string
}

export type Mission = {
  mission_name: string
  // id: number
  // name: string
}
// export type Mission = {
//   description: string
//   id: ID
//   manufacturers: [string]
//   name: string
//   twitter: string
//   website: string
//   wikipedia: string
// }
