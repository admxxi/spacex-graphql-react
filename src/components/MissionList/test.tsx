import React from 'react'
import { MissionList } from './index'
import { render, screen } from '@testing-library/react'
import { Launch as Ilaunch } from '../../@types/graphql'

const MockMissions: Ilaunch[] = [
  {
    id: '1',
    is_tentative: false,
    upcoming: false,
    mission_name: 'Dummy Mission 1',
    links: {
      __typename: 'LaunchLinks',
      article_link: 'https://dummy.article.com/',
      video_link: 'https://dummy.video.com/',
      flickr_images: [
        'https://live.staticflickr.com/65535/50618376646_8f52c31fc4_o.jpg'
      ],
      mission_patch: 'https://i.imgur.com/t5R4BAQ.png'
    },
    launch_date_utc: '2020-11-16T00:27:00.000Z',
    details: 'dummy details'
  },
  {
    id: '2',
    is_tentative: false,
    upcoming: false,
    mission_name: 'Dummy Mission 2',
    links: {
      __typename: 'LaunchLinks',
      article_link: 'https://dummy.article.com/',
      video_link: 'https://dummy.video.com/',
      flickr_images: [
        'https://live.staticflickr.com/65535/50618376646_8f52c31fc4_o.jpg'
      ],
      mission_patch: 'https://i.imgur.com/t5R4BAQ.png'
    },
    launch_date_utc: '2020-11-16T00:27:00.000Z',
    details: 'dummy details'
  }
]

it('renders correctly', () => {
  const { container } = render(<MissionList missions={MockMissions} />)
  expect(container.firstChild).toMatchSnapshot()
})

it('renders number of missions correctly', async () => {
  const { debug } = render(<MissionList missions={MockMissions} />)
  const myElement = await screen.findAllByText(/Dummy Mission/i)
  expect(myElement).toHaveLength(2)
})
