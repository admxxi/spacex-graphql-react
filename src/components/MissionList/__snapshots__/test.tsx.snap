// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="MissionList__Style-sc-1d8oiwg-0 cBukNc"
>
  <h1
    class="title is-2 my-3"
  >
    Last Launches ðŸš€
  </h1>
  <div
    class="container"
  >
    <div
      class="Mission__Style-wzd66q-0 lhGnqF"
    >
      <article
        class="card"
      >
        <div
          class="card-content"
        >
          <div
            class="media is-align-content-center"
          >
            <div
              class="media-content"
            >
              <a
                class="is-link is-3"
              >
                Dummy Mission 1
              </a>
              <p
                class="subtitle is-4 mt-2"
              >
                dummy details
              </p>
              <div
                class="ma-1 subtitle is-5 is-italic has-text-right mx-1"
              >
                11/16/2020
              </div>
            </div>
          </div>
        </div>
      </article>
    </div>
    <div
      class="Mission__Style-wzd66q-0 lhGnqF"
    >
      <article
        class="card"
      >
        <div
          class="card-content"
        >
          <div
            class="media is-align-content-center"
          >
            <div
              class="media-content"
            >
              <a
                class="is-link is-3"
              >
                Dummy Mission 2
              </a>
              <p
                class="subtitle is-4 mt-2"
              >
                dummy details
              </p>
              <div
                class="ma-1 subtitle is-5 is-italic has-text-right mx-1"
              >
                11/16/2020
              </div>
            </div>
          </div>
        </div>
      </article>
    </div>
  </div>
</div>
`;
